### DISCUSSION TITLE
Adding deformation gradient to AD version of computeStrain classes

### DISCUSSION THREAD

**Discussion by srinath-chakravarthy** [idaholab/moose#26381] [Q&A Modules: Solid mechanics]
Hello all, 
deformation_gradient is not available for the AD versions of the computeStrain classes but seem to be available to the non-AD versions. Any chance this can be included into the tensor_mechanics classes. I have a local implementation that i can perhaps spruce up and submit as a PR if necessary

**Comment by GiudGiud** [comment]
@dschwen @gambka

**Reply by srinath-chakravarthy** [reply]
Any update on this ?

**Reply by lynnmunday** [reply]
Do you have a PR where you add something like this to ADComputeIncrementalStrainBase.h:
```
MaterialProperty<RankTwoTensor> & _deformation_gradient;
```
and then copy in the all the code needed to compute _deformation_gradient[_qp] from ComputeFiniteStrain.C into ADComputeFiniteStrain.C?

**Comment by srinath-chakravarthy** [comment]
I do not have a PR, but i do have the code. Let me know what i need to do.

**Reply by lynnmunday** [reply]
I'm not sure how familiar you are with the moose process for merging code changes, there is probably a better explanation of this somewhere in the documentation or training but you need to do something like the following:

First submit an issue to MOOSE for this enhancement using the github tab, you can look at a few others to see the format.  You need to create your own fork of moose.  Then you will create a branch on your fork of moose, push that branch to your github account and then create a pull request from your fork onto the inl labs moose repository.

**Reply by lynnmunday** [reply]
Then someone will review and merge it into the main repository.

**Reply by GiudGiud** [reply]
this is the page on how to contribute to MOOSE
https://mooseframework.inl.gov/framework_development/index.html

**Comment by srinath-chakravarthy** [comment]
Fantastic,
Will work on this over the next few days and submit.

Cheers
Srinath
